fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/LogicalMetric.java

```diff
@@ -80,6 +80,28 @@ public LogicalMetric(TemplateDruidQuery templateDruidQuery, ResultSetMapper calc
         this(templateDruidQuery, calculation, name, name, DEFAULT_CATEGORY, name);
     }
 
+    /**
+     * Constructor. Builds a Logical Metric whose instance variables are provided by a LogicalMetricInfo object.
+     *
+     * @param templateDruidQuery  Query the metric needs
+     * @param calculation  Mapper for the metric
+     * @param logicalMetricInfo  Logical Metric info provider
+     */
+    public LogicalMetric(
```

Comment:
Also, would we want to mark the other constructors with `@Deprecated` telling people to use a `LogicalMetricInfo`?